### ISSUE TITLE
Incorrect Style/SafeNavigation autofix with ! and method chain

### ISSUE THREAD

**Issue by ryym** [bbatsov/rubocop#4765]
Hi, I found a problem similar to #4745.
`rubocop -a` fixes `Style/SafeNavigation` error incorrectly.

Sample code (sample.rb):

```ruby
# frozen_string_literal: true

def decorate(value)
  return value unless value && !value.downcase.include?('decorated')
  "decorated #{value}"
end
```

--------

## Expected behavior

`rubocop -a sample.rb` does nothing. In this case `decorate(nil)` returns nil.

## Actual behavior

It fixes the code like below:

```ruby
# frozen_string_literal: true

def decorate(value)
  return value if value&.downcase&.include?('decorated')
  "decorated #{value}"
end
```

This changes the `decorate` method behavior. Now `decorate(nil)` returns `"decorated "` instead of nil.

## Steps to reproduce the problem

Run `rubocop -a sample.rb` using this `.rubocop.yml`:

```yml
AllCops:
  TargetRubyVersion: 2.4
```

## RuboCop version

```
$ rubocop -V
0.50.0 (using Parser 2.4.0.0, running on ruby 2.4.1 x86_64-darwin16)
```

I also checked using the latest commit 3920afac but it still happened.
However the latest commit version does nothing If `.downcase` does not exist.

**Comment by Drenmi** [comment]
This will be fixed in the next release through #4752. 🙂

**Comment by Drenmi** [comment]
There's a change in #4752, that will prevent the cop from doing anything on a negated expression.

We could potentially try to do the suggested correction, but I think it opens up a lot of edge cases and will force more complexity (it's already pretty bad) into the cop.

**Comment by ryym** [comment]
Right, it seems difficult to implement the `Style/SafeNavigation` cop perfectly 😓 

I tried to fix the cop and solved the above problem. https://github.com/ryym/rubocop/commit/84d64f769b8bdc6a070218b1e58fa29d8de7a59c
but I'm not confident that this fix could handle all negated expressions properly, and that it does not affect other cases.

**Comment by Drenmi** [comment]
Fixed in #4752 and #4749.

**Comment by ryym** [comment]
Thank you 🎉
